diff -r -U 3 minecraft/net/minecraft/client/controller/PlayerControllerCreative.java minecraft_patched/net/minecraft/client/controller/PlayerControllerCreative.java
--- minecraft/net/minecraft/client/controller/PlayerControllerCreative.java	2021-11-27 12:59:18.336358100 +0200
+++ minecraft_patched/net/minecraft/client/controller/PlayerControllerCreative.java	2021-11-27 12:50:41.588801900 +0200
@@ -1,11 +1,17 @@
 package net.minecraft.client.controller;
 
+import net.minecraft.client.Minecraft;
 import net.minecraft.client.Session;
 import net.minecraft.game.entity.player.EntityPlayer;
 import net.minecraft.game.item.ItemStack;
 import net.minecraft.game.world.block.Block;
 
 public final class PlayerControllerCreative extends PlayerController {
+	public PlayerControllerCreative(Minecraft var1) {
+		super(var1);
+		this.isInTestMode = true;
+	}
+
 	public final void onRespawn(EntityPlayer var1) {
 		for(int var2 = 0; var2 < 9; ++var2) {
 			if(var1.inventory.mainInventory[var2] == null) {
diff -r -U 3 minecraft/net/minecraft/client/gui/GuiIngame.java minecraft_patched/net/minecraft/client/gui/GuiIngame.java
--- minecraft/net/minecraft/client/gui/GuiIngame.java	2021-11-27 12:59:18.575371800 +0200
+++ minecraft_patched/net/minecraft/client/gui/GuiIngame.java	2021-11-27 12:58:06.891271700 +0200
@@ -127,26 +127,25 @@
 		GL11.glPopMatrix();
 
 		for(var10 = 0; var10 < 9; ++var10) {
-			float var25 = (float)(var3 / 2 - 90 + var10 * 20 + 2);
-			var12 = var19 - 16 - 3;
-			float var21 = var25;
+			int var25 = var3 / 2 - 90 + var10 * 20 + 2;
+			int var21 = var19 - 16 - 3;
 			ItemStack var22;
 			if((var22 = this.mc.thePlayer.inventory.mainInventory[var10]) != null) {
 				float var9;
 				if((var9 = (float)var22.animationsToGo - var1) > 0.0F) {
 					GL11.glPushMatrix();
-					var25 = 1.0F + var9 / 5.0F;
-					GL11.glTranslatef((float)(var21 + 8), (float)(var12 + 12), 0.0F);
-					GL11.glScalef(1.0F / var25, (var25 + 1.0F) / 2.0F, 1.0F);
-					GL11.glTranslatef((float)(-(var21 + 8)), (float)(-(var12 + 12)), 0.0F);
+					float var24 = 1.0F + var9 / 5.0F;
+					GL11.glTranslatef((float)(var25 + 8), (float)(var21 + 12), 0.0F);
+					GL11.glScalef(1.0F / var24, (var24 + 1.0F) / 2.0F, 1.0F);
+					GL11.glTranslatef((float)(-(var25 + 8)), (float)(-(var21 + 12)), 0.0F);
 				}
 
-				itemRenderer.doRender(this.mc.renderEngine, var22, (int)var25, var12);
+				itemRenderer.doRender(this.mc.renderEngine, var22, (int)var25, var21);
 				if(var9 > 0.0F) {
 					GL11.glPopMatrix();
 				}
 
-				itemRenderer.renderItemOverlayIntoGUI(this.mc.fontRenderer, var22, (int)var25, var12);
+				itemRenderer.renderItemOverlayIntoGUI(this.mc.fontRenderer, var22, (int)var25, var21);
 			}
 		}
 
diff -r -U 3 minecraft/net/minecraft/client/render/entity/RenderPlayer.java minecraft_patched/net/minecraft/client/render/entity/RenderPlayer.java
--- minecraft/net/minecraft/client/render/entity/RenderPlayer.java	2021-11-27 12:59:18.462365300 +0200
+++ minecraft_patched/net/minecraft/client/render/entity/RenderPlayer.java	2021-11-27 12:51:26.187352700 +0200
@@ -41,8 +41,8 @@
 		if((var6 = var5.inventory.armorInventory[var4]) != null && (var7 = var6.getItem()) instanceof ItemArmor) {
 			ItemArmor var8 = (ItemArmor)var7;
 			this.loadTexture("/armor/" + armorFilenamePrefix[var8.renderIndex] + "_" + (var3 == 2 ? 2 : 1) + ".png");
-			ModelBiped var9;
-			(var9 = var3 == 2 ? this.modelArmor : this.modelArmorChestplate).bipedHead.showModel = var3 == 0;
+			ModelBiped var9 = var3 == 2 ? this.modelArmor : this.modelArmorChestplate;
+			var9.bipedHead.showModel = var3 == 0;
 			var9.bipedHeadwear.showModel = var3 == 0;
 			var9.bipedBody.showModel = var3 == 1 || var3 == 2;
 			var9.bipedRightArm.showModel = var3 == 1;
diff -r -U 3 minecraft/net/minecraft/client/render/EntityRenderer.java minecraft_patched/net/minecraft/client/render/EntityRenderer.java
--- minecraft/net/minecraft/client/render/EntityRenderer.java	2021-11-27 12:59:18.594372900 +0200
+++ minecraft_patched/net/minecraft/client/render/EntityRenderer.java	2021-11-27 12:51:10.101432700 +0200
@@ -346,10 +346,10 @@
 				double var33 = (double)(MathHelper.cos(var58.rotationYaw / 180.0F * 3.1415927F) * MathHelper.cos(var58.rotationPitch / 180.0F * 3.1415927F)) * 4.0D;
 				var35 = (double)(-MathHelper.sin(var58.rotationPitch / 180.0F * 3.1415927F)) * 4.0D;
 
-				for(var46 = (float)false; var46 < 8; ++var46) {
-					float var49 = (float)(((var46 & 1) << 1) - 1);
-					float var44 = (float)(((var46 >> 1 & 1) << 1) - 1);
-					var10 = (float)(((var46 >> 2 & 1) << 1) - 1);
+				for(int var47 = 0; var46 < 8; ++var47) {
+					float var49 = (float)(((var47 & 1) << 1) - 1);
+					float var44 = (float)(((var47 >> 1 & 1) << 1) - 1);
+					var10 = (float)(((var47 >> 2 & 1) << 1) - 1);
 					var49 *= 0.1F;
 					var44 *= 0.1F;
 					var10 *= 0.1F;
diff -r -U 3 minecraft/net/minecraft/game/entity/animal/EntityPig.java minecraft_patched/net/minecraft/game/entity/animal/EntityPig.java
--- minecraft/net/minecraft/game/entity/animal/EntityPig.java	2021-11-27 12:59:18.482366500 +0200
+++ minecraft_patched/net/minecraft/game/entity/animal/EntityPig.java	2021-11-27 12:51:45.682467800 +0200
@@ -1,8 +1,13 @@
 package net.minecraft.game.entity.animal;
 
 import net.minecraft.game.item.Item;
+import net.minecraft.game.world.World;
 
 public class EntityPig extends EntityAnimal {
+	public EntityPig(World var1) {
+		super(var1);
+	}
+
 	protected final String getLivingSound() {
 		return "mob.pig";
 	}
diff -r -U 3 minecraft/net/minecraft/game/entity/animal/EntitySheep.java minecraft_patched/net/minecraft/game/entity/animal/EntitySheep.java
--- minecraft/net/minecraft/game/entity/animal/EntitySheep.java	2021-11-27 12:59:18.483366500 +0200
+++ minecraft_patched/net/minecraft/game/entity/animal/EntitySheep.java	2021-11-27 12:51:52.609864000 +0200
@@ -3,9 +3,14 @@
 import net.minecraft.game.entity.Entity;
 import net.minecraft.game.entity.EntityLiving;
 import net.minecraft.game.entity.misc.EntityItem;
+import net.minecraft.game.world.World;
 import net.minecraft.game.world.block.Block;
 
 public class EntitySheep extends EntityAnimal {
+	public EntitySheep(World var1) {
+		super(var1);
+	}
+
 	public boolean sheared;
 
 	public final boolean attackEntityFrom(Entity var1, int var2) {
diff -r -U 3 minecraft/net/minecraft/game/entity/monster/EntityCreeper.java minecraft_patched/net/minecraft/game/entity/monster/EntityCreeper.java
--- minecraft/net/minecraft/game/entity/monster/EntityCreeper.java	2021-11-27 12:59:18.486366700 +0200
+++ minecraft_patched/net/minecraft/game/entity/monster/EntityCreeper.java	2021-11-27 12:52:06.124637000 +0200
@@ -2,8 +2,13 @@
 
 import net.minecraft.game.entity.Entity;
 import net.minecraft.game.item.Item;
+import net.minecraft.game.world.World;
 
 public class EntityCreeper extends EntityMob {
+	public EntityCreeper(World var1) {
+		super(var1);
+	}
+
 	private int timeSinceIgnited;
 	private int lastActiveTime;
 	private int fuseDuration;
diff -r -U 3 minecraft/net/minecraft/game/entity/monster/EntityGiantZombie.java minecraft_patched/net/minecraft/game/entity/monster/EntityGiantZombie.java
--- minecraft/net/minecraft/game/entity/monster/EntityGiantZombie.java	2021-11-27 12:59:18.486366700 +0200
+++ minecraft_patched/net/minecraft/game/entity/monster/EntityGiantZombie.java	2021-11-27 12:52:13.419054200 +0200
@@ -1,6 +1,12 @@
 package net.minecraft.game.entity.monster;
 
+import net.minecraft.game.world.World;
+
 public class EntityGiantZombie extends EntityMob {
+	public EntityGiantZombie(World var1) {
+		super(var1);
+	}
+
 	protected final float getBlockPathWeight(int var1, int var2, int var3) {
 		return this.worldObj.getBrightness(var1, var2, var3) - 0.5F;
 	}
diff -r -U 3 minecraft/net/minecraft/game/entity/monster/EntitySkeleton.java minecraft_patched/net/minecraft/game/entity/monster/EntitySkeleton.java
--- minecraft/net/minecraft/game/entity/monster/EntitySkeleton.java	2021-11-27 12:59:18.488366800 +0200
+++ minecraft_patched/net/minecraft/game/entity/monster/EntitySkeleton.java	2021-11-27 12:52:21.410511300 +0200
@@ -3,9 +3,14 @@
 import net.minecraft.game.entity.Entity;
 import net.minecraft.game.entity.projectile.EntityArrow;
 import net.minecraft.game.item.Item;
+import net.minecraft.game.world.World;
 import util.MathHelper;
 
 public class EntitySkeleton extends EntityMob {
+	public EntitySkeleton(World var1) {
+		super(var1);
+	}
+
 	public final void updatePlayerActionState() {
 		super.updatePlayerActionState();
 	}
diff -r -U 3 minecraft/net/minecraft/game/entity/monster/EntitySpider.java minecraft_patched/net/minecraft/game/entity/monster/EntitySpider.java
--- minecraft/net/minecraft/game/entity/monster/EntitySpider.java	2021-11-27 12:59:18.489366900 +0200
+++ minecraft_patched/net/minecraft/game/entity/monster/EntitySpider.java	2021-11-27 12:52:27.042833500 +0200
@@ -2,9 +2,14 @@
 
 import net.minecraft.game.entity.Entity;
 import net.minecraft.game.item.Item;
+import net.minecraft.game.world.World;
 import util.MathHelper;
 
 public class EntitySpider extends EntityMob {
+	public EntitySpider(World var1) {
+		super(var1);
+	}
+
 	protected final Entity findPlayerToAttack() {
 		return this.getBrightness(1.0F) < 0.5F && this.worldObj.playerEntity.getDistanceToEntity(this) < 256.0D ? this.worldObj.playerEntity : null;
 	}
diff -r -U 3 minecraft/net/minecraft/game/entity/monster/EntityZombie.java minecraft_patched/net/minecraft/game/entity/monster/EntityZombie.java
--- minecraft/net/minecraft/game/entity/monster/EntityZombie.java	2021-11-27 12:59:18.491367000 +0200
+++ minecraft_patched/net/minecraft/game/entity/monster/EntityZombie.java	2021-11-27 12:52:32.082121700 +0200
@@ -1,8 +1,13 @@
 package net.minecraft.game.entity.monster;
 
 import net.minecraft.game.item.Item;
+import net.minecraft.game.world.World;
 
 public class EntityZombie extends EntityMob {
+	public EntityZombie(World var1) {
+		super(var1);
+	}
+
 	public final void updatePlayerActionState() {
 		super.updatePlayerActionState();
 	}
diff -r -U 3 minecraft/net/minecraft/game/item/Item.java minecraft_patched/net/minecraft/game/item/Item.java
--- minecraft/net/minecraft/game/item/Item.java	2021-11-27 12:59:18.494367200 +0200
+++ minecraft_patched/net/minecraft/game/item/Item.java	2021-11-27 12:53:22.934030300 +0200
@@ -135,335 +135,335 @@
 		ItemSpade var10000 = new ItemSpade(0, 2);
 		byte var1 = 82;
 		ItemSpade var0 = var10000;
-		var10000.iconIndex = var1;
+		var10000.setIconIndex(var1);
 		shovel = var0;
 		ItemPickaxe var15 = new ItemPickaxe(1, 2);
 		var1 = 98;
 		ItemPickaxe var2 = var15;
-		var15.iconIndex = var1;
+		var15.setIconIndex(var1);
 		pickaxeSteel = var2;
 		ItemAxe var16 = new ItemAxe(2, 2);
 		var1 = 114;
 		ItemAxe var3 = var16;
-		var16.iconIndex = var1;
+		var16.setIconIndex(var1);
 		axeSteel = var3;
 		ItemFlintAndSteel var17 = new ItemFlintAndSteel(3);
 		var1 = 5;
 		ItemFlintAndSteel var4 = var17;
-		var17.iconIndex = var1;
+		var17.setIconIndex(var1);
 		striker = var4;
 		ItemFood var18 = new ItemFood(4, 4);
 		var1 = 10;
 		ItemFood var5 = var18;
-		var18.iconIndex = var1;
+		var18.setIconIndex(var1);
 		appleRed = var5;
 		ItemBow var19 = new ItemBow(5);
 		var1 = 21;
 		ItemBow var6 = var19;
-		var19.iconIndex = var1;
+		var19.setIconIndex(var1);
 		bow = var6;
 		Item var20 = new Item(6);
 		var1 = 37;
 		Item var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		arrow = var7;
 		var20 = new Item(7);
 		var1 = 7;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		coal = var7;
 		var20 = new Item(8);
 		var1 = 55;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		diamond = var7;
 		var20 = new Item(9);
 		var1 = 23;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		ingotIron = var7;
 		var20 = new Item(10);
 		var1 = 39;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		ingotGold = var7;
 		ItemSword var21 = new ItemSword(11, 2);
 		var1 = 66;
 		ItemSword var8 = var21;
-		var21.iconIndex = var1;
+		var21.setIconIndex(var1);
 		swordSteel = var8;
 		var21 = new ItemSword(12, 0);
 		var1 = 64;
 		var8 = var21;
-		var21.iconIndex = var1;
+		var21.setIconIndex(var1);
 		swordWood = var8;
 		var10000 = new ItemSpade(13, 0);
 		var1 = 80;
 		var0 = var10000;
-		var10000.iconIndex = var1;
+		var10000.setIconIndex(var1);
 		shovelWood = var0;
 		var15 = new ItemPickaxe(14, 0);
 		var1 = 96;
 		var2 = var15;
-		var15.iconIndex = var1;
+		var15.setIconIndex(var1);
 		pickaxeWood = var2;
 		var16 = new ItemAxe(15, 0);
 		var1 = 112;
 		var3 = var16;
-		var16.iconIndex = var1;
+		var16.setIconIndex(var1);
 		axeWood = var3;
 		var21 = new ItemSword(16, 1);
 		var1 = 65;
 		var8 = var21;
-		var21.iconIndex = var1;
+		var21.setIconIndex(var1);
 		swordStone = var8;
 		var10000 = new ItemSpade(17, 1);
 		var1 = 81;
 		var0 = var10000;
-		var10000.iconIndex = var1;
+		var10000.setIconIndex(var1);
 		shovelStone = var0;
 		var15 = new ItemPickaxe(18, 1);
 		var1 = 97;
 		var2 = var15;
-		var15.iconIndex = var1;
+		var15.setIconIndex(var1);
 		pickaxeStone = var2;
 		var16 = new ItemAxe(19, 1);
 		var1 = 113;
 		var3 = var16;
-		var16.iconIndex = var1;
+		var16.setIconIndex(var1);
 		axeStone = var3;
 		var21 = new ItemSword(20, 3);
 		var1 = 67;
 		var8 = var21;
-		var21.iconIndex = var1;
+		var21.setIconIndex(var1);
 		swordDiamond = var8;
 		var10000 = new ItemSpade(21, 3);
 		var1 = 83;
 		var0 = var10000;
-		var10000.iconIndex = var1;
+		var10000.setIconIndex(var1);
 		shovelDiamond = var0;
 		var15 = new ItemPickaxe(22, 3);
 		var1 = 99;
 		var2 = var15;
-		var15.iconIndex = var1;
+		var15.setIconIndex(var1);
 		pickaxeDiamond = var2;
 		var16 = new ItemAxe(23, 3);
 		var1 = 115;
 		var3 = var16;
-		var16.iconIndex = var1;
+		var16.setIconIndex(var1);
 		axeDiamond = var3;
 		var20 = new Item(24);
 		var1 = 53;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		stick = var7;
 		var20 = new Item(25);
 		var1 = 71;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		bowlEmpty = var7;
 		ItemSoup var22 = new ItemSoup(26, 10);
 		var1 = 72;
 		ItemSoup var9 = var22;
-		var22.iconIndex = var1;
+		var22.setIconIndex(var1);
 		bowlSoup = var9;
 		var21 = new ItemSword(27, 0);
 		var1 = 68;
 		var8 = var21;
-		var21.iconIndex = var1;
+		var21.setIconIndex(var1);
 		swordGold = var8;
 		var10000 = new ItemSpade(28, 0);
 		var1 = 84;
 		var0 = var10000;
-		var10000.iconIndex = var1;
+		var10000.setIconIndex(var1);
 		shovelGold = var0;
 		var15 = new ItemPickaxe(29, 0);
 		var1 = 100;
 		var2 = var15;
-		var15.iconIndex = var1;
+		var15.setIconIndex(var1);
 		pickaxeGold = var2;
 		var16 = new ItemAxe(30, 0);
 		var1 = 116;
 		var3 = var16;
-		var16.iconIndex = var1;
+		var16.setIconIndex(var1);
 		axeGold = var3;
 		var20 = new Item(31);
 		var1 = 8;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		silk = var7;
 		var20 = new Item(32);
 		var1 = 24;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		feather = var7;
 		var20 = new Item(33);
 		var1 = 40;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		gunpowder = var7;
 		ItemHoe var23 = new ItemHoe(34, 0);
 		short var11 = 128;
 		ItemHoe var10 = var23;
-		var23.iconIndex = var11;
+		var23.setIconIndex(var11);
 		hoeWood = var10;
 		var23 = new ItemHoe(35, 1);
 		var11 = 129;
 		var10 = var23;
-		var23.iconIndex = var11;
+		var23.setIconIndex(var11);
 		hoeStone = var10;
 		var23 = new ItemHoe(36, 2);
 		var11 = 130;
 		var10 = var23;
-		var23.iconIndex = var11;
+		var23.setIconIndex(var11);
 		hoeSteel = var10;
 		var23 = new ItemHoe(37, 3);
 		var11 = 131;
 		var10 = var23;
-		var23.iconIndex = var11;
+		var23.setIconIndex(var11);
 		hoeDiamond = var10;
 		var23 = new ItemHoe(38, 4);
 		var11 = 132;
 		var10 = var23;
-		var23.iconIndex = var11;
+		var23.setIconIndex(var11);
 		hoeGold = var10;
 		ItemSeeds var24 = new ItemSeeds(39, Block.blockDiamond.blockID);
 		var1 = 9;
 		ItemSeeds var12 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		seeds = var12;
 		var20 = new Item(40);
 		var1 = 25;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		wheat = var7;
 		var18 = new ItemFood(41, 5);
 		var1 = 41;
 		var5 = var18;
-		var18.iconIndex = var1;
+		var18.setIconIndex(var1);
 		bread = var5;
 		ItemArmor var25 = new ItemArmor(42, 0, 0, 0);
 		var1 = 0;
 		ItemArmor var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		helmetLeather = var13;
 		var25 = new ItemArmor(43, 0, 0, 1);
 		var1 = 16;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		plateLeather = var13;
 		var25 = new ItemArmor(44, 0, 0, 2);
 		var1 = 32;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		legsLeather = var13;
 		var25 = new ItemArmor(45, 0, 0, 3);
 		var1 = 48;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		bootsLeather = var13;
 		var25 = new ItemArmor(46, 1, 1, 0);
 		var1 = 1;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		helmetChain = var13;
 		var25 = new ItemArmor(47, 1, 1, 1);
 		var1 = 17;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		plateChain = var13;
 		var25 = new ItemArmor(48, 1, 1, 2);
 		var1 = 33;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		legsChain = var13;
 		var25 = new ItemArmor(49, 1, 1, 3);
 		var1 = 49;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		bootsChain = var13;
 		var25 = new ItemArmor(50, 2, 2, 0);
 		var1 = 2;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		helmetSteel = var13;
 		var25 = new ItemArmor(51, 2, 2, 1);
 		var1 = 18;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		plateSteel = var13;
 		var25 = new ItemArmor(52, 2, 2, 2);
 		var1 = 34;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		legsSteel = var13;
 		var25 = new ItemArmor(53, 2, 2, 3);
 		var1 = 50;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		bootsSteel = var13;
 		var25 = new ItemArmor(54, 3, 3, 0);
 		var1 = 3;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		helmetDiamond = var13;
 		var25 = new ItemArmor(55, 3, 3, 1);
 		var1 = 19;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		plateDiamond = var13;
 		var25 = new ItemArmor(56, 3, 3, 2);
 		var1 = 35;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		legsDiamond = var13;
 		var25 = new ItemArmor(57, 3, 3, 3);
 		var1 = 51;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		bootsDiamond = var13;
 		var25 = new ItemArmor(58, 1, 4, 0);
 		var1 = 4;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		helmetGold = var13;
 		var25 = new ItemArmor(59, 1, 4, 1);
 		var1 = 20;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		plateGold = var13;
 		var25 = new ItemArmor(60, 1, 4, 2);
 		var1 = 36;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		legsGold = var13;
 		var25 = new ItemArmor(61, 1, 4, 3);
 		var1 = 52;
 		var13 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		bootsGold = var13;
 		var20 = new Item(62);
 		var1 = 6;
 		var7 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		flint = var7;
 		var18 = new ItemFood(63, 3);
 		var1 = 87;
 		var5 = var18;
-		var18.iconIndex = var1;
+		var18.setIconIndex(var1);
 		porkRaw = var5;
 		var18 = new ItemFood(64, 8);
 		var1 = 88;
-		var18.iconIndex = var1;
+		var18.setIconIndex(var1);
 		ItemPainting var26 = new ItemPainting(65);
 		var1 = 26;
 		ItemPainting var14 = var26;
-		var26.iconIndex = var1;
+		var26.setIconIndex(var1);
 		painting = var14;
 		var18 = new ItemFood(66, 42);
 		var1 = 11;
 		var5 = var18;
-		var18.iconIndex = var1;
+		var18.setIconIndex(var1);
 		appleGold = var5;
 	}
 }
diff -r -U 3 minecraft/net/minecraft/game/world/path/Pathfinder.java minecraft_patched/net/minecraft/game/world/path/Pathfinder.java
--- minecraft/net/minecraft/game/world/path/Pathfinder.java	2021-11-27 12:59:18.541369800 +0200
+++ minecraft_patched/net/minecraft/game/world/path/Pathfinder.java	2021-11-27 12:52:54.428399800 +0200
@@ -27,7 +27,7 @@
 	private PathEntity addToPath(Entity var1, double var2, double var4, double var6, float var8) {
 		this.path.clearPath();
 		this.pointMap.clear();
-		Object var9 = this.openPoint((int)var1.boundingBox.minX, (int)var1.boundingBox.minY, (int)var1.boundingBox.minZ);
+		PathPoint var9 = this.openPoint((int)var1.boundingBox.minX, (int)var1.boundingBox.minY, (int)var1.boundingBox.minZ);
 		PathPoint var21 = this.openPoint((int)(var2 - (double)(var1.width / 2.0F)), (int)var4, (int)(var6 - (double)(var1.width / 2.0F)));
 		PathPoint var3 = new PathPoint((int)(var1.width + 1.0F), (int)(var1.height + 1.0F), (int)(var1.width + 1.0F));
 		float var24 = var8;
@@ -60,7 +60,7 @@
 			}
 
 			var25.isFirst = true;
-			Object var15 = 0;
+			int var15 = 0;
 			byte var16 = 0;
 			if(var20.getVerticalOffset(var25.xCoord, var25.yCoord + 1, var25.zCoord, var5) > 0) {
 				var16 = 1;
@@ -87,9 +87,7 @@
 				var20.pathOptions[var15++] = var10;
 			}
 
-			var9 = var15;
-
-			for(int var26 = 0; var26 < var9; ++var26) {
+			for(int var26 = 0; var26 < var15; ++var26) {
 				PathPoint var11 = var20.pathOptions[var26];
 				float var12 = var25.totalPathDistance + var25.distanceTo(var11);
 				if(!var11.isAssigned() || var12 < var11.totalPathDistance) {
