diff -r -U 3 minecraft/net/minecraft/client/render/entity/RenderPlayer.java minecraft_patched/net/minecraft/client/render/entity/RenderPlayer.java
--- minecraft/net/minecraft/client/render/entity/RenderPlayer.java	2021-11-26 21:41:39.163573600 +0200
+++ minecraft_patched/net/minecraft/client/render/entity/RenderPlayer.java	2021-11-26 21:37:12.053295800 +0200
@@ -41,8 +41,8 @@
 		if((var6 = var5.inventory.armorInventory[var4]) != null && (var7 = var6.getItem()) instanceof ItemArmor) {
 			ItemArmor var8 = (ItemArmor)var7;
 			this.loadTexture("/armor/" + armorFilenamePrefix[var8.renderIndex] + "_" + (var3 == 2 ? 2 : 1) + ".png");
-			ModelBiped var9;
-			(var9 = var3 == 2 ? this.modelArmor : this.modelArmorChestplate).bipedHead.showModel = var3 == 0;
+			ModelBiped var9 = var3 == 2 ? this.modelArmor : this.modelArmorChestplate;
+			var9.bipedHead.showModel = var3 == 0;
 			var9.bipedHeadwear.showModel = var3 == 0;
 			var9.bipedBody.showModel = var3 == 1 || var3 == 2;
 			var9.bipedRightArm.showModel = var3 == 1;
diff -r -U 3 minecraft/net/minecraft/client/gui/GuiIngame.java minecraft_patched/net/minecraft/client/gui/GuiIngame.java
--- minecraft/net/minecraft/client/gui/GuiIngame.java	2021-11-27 12:10:47.459865500 +0200
+++ minecraft_patched/net/minecraft/client/gui/GuiIngame.java	2021-11-27 12:09:11.930401500 +0200
@@ -158,26 +158,25 @@
 		GL11.glPopMatrix();
 
 		for(var10 = 0; var10 < 9; ++var10) {
-			float var26 = (float)(var3 / 2 - 90 + var10 * 20 + 2);
-			var12 = var19 - 16 - 3;
-			float var20 = var26;
+			int var26 = var3 / 2 - 90 + var10 * 20 + 2;
+			int var20 = var19 - 16 - 3;
 			ItemStack var23;
 			if((var23 = this.mc.thePlayer.inventory.mainInventory[var10]) != null) {
 				float var24;
 				if((var24 = (float)var23.animationsToGo - var1) > 0.0F) {
 					GL11.glPushMatrix();
-					var26 = 1.0F + var24 / 5.0F;
-					GL11.glTranslatef((float)(var20 + 8), (float)(var12 + 12), 0.0F);
-					GL11.glScalef(1.0F / var26, (var26 + 1.0F) / 2.0F, 1.0F);
-					GL11.glTranslatef((float)(-(var20 + 8)), (float)(-(var12 + 12)), 0.0F);
+					float var25 = 1.0F + var24 / 5.0F;
+					GL11.glTranslatef((float)(var26 + 8), (float)(var20 + 12), 0.0F);
+					GL11.glScalef(1.0F / var25, (var25 + 1.0F) / 2.0F, 1.0F);
+					GL11.glTranslatef((float)(-(var26 + 8)), (float)(-(var20 + 12)), 0.0F);
 				}
 
-				itemRenderer.renderItemIntoGUI(this.mc.renderEngine, var23, (int)var26, var12);
+				itemRenderer.renderItemIntoGUI(this.mc.renderEngine, var23, var26, var20);
 				if(var24 > 0.0F) {
 					GL11.glPopMatrix();
 				}
 
-				itemRenderer.renderItemOverlayIntoGUI(this.mc.fontRenderer, var23, (int)var26, var12);
+				itemRenderer.renderItemOverlayIntoGUI(this.mc.fontRenderer, var23, var26, var20);
 			}
 		}
 
diff -r -U 3 minecraft/net/minecraft/client/ThreadDownloadResources.java minecraft_patched/net/minecraft/client/ThreadDownloadResources.java
--- minecraft/net/minecraft/client/ThreadDownloadResources.java	2021-11-26 19:31:50.992590200 +0200
+++ minecraft_patched/net/minecraft/client/ThreadDownloadResources.java	2021-11-26 20:24:37.068679400 +0200
@@ -1,11 +1,14 @@
 package net.minecraft.client;
 
+import java.io.BufferedReader;
 import java.io.DataInputStream;
 import java.io.DataOutputStream;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
+import java.io.InputStreamReader;
 import java.net.URL;
+import java.util.ArrayList;
 
 public final class ThreadDownloadResources extends Thread {
 	private File resourcesFolder;
@@ -23,7 +26,63 @@
 	}
 
 	public final void run() {
-		// $FF: Couldn't be decompiled
+		try {
+			final ArrayList<String> list = new ArrayList<String>();
+			final URL url = new URL("http://www.minecraft.net/resources/");
+			final BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(url.openStream()));
+			String line;
+			while ((line = bufferedReader.readLine()) != null) {
+				list.add(line);
+			}
+			bufferedReader.close();
+			for (int i = 0; i < list.size(); ++i) {
+				final URL url2 = url;
+				final String s = list.get(i);
+				final URL context = url2;
+				Label_0334: {
+					try {
+						final String[] split;
+						final String child = (split = s.split(","))[0];
+						final int int1 = Integer.parseInt(split[1]);
+						Long.parseLong(split[2]);
+						final File file;
+						if (!(file = new File(this.resourcesFolder, child)).exists() || file.length() != int1) {
+							file.getParentFile().mkdirs();
+							this.downloadResource(new URL(context, child.replaceAll(" ", "%20")), file);
+							if (this.closing) {
+								break Label_0334;
+							}
+						}
+						final Minecraft mc = this.mc;
+						final String s2 = child;
+						final File file2 = file;
+						final String s3 = s2;
+						final Minecraft minecraft = mc;
+						final int index = s3.indexOf("/");
+						final String substring = s3.substring(0, index);
+						final String substring2 = s3.substring(index + 1);
+						if (substring.equalsIgnoreCase("sound")) {
+							minecraft.sndManager.addSound(substring2, file2);
+						}
+						else if (substring.equalsIgnoreCase("newsound")) {
+							minecraft.sndManager.addSound(substring2, file2);
+						}
+						else if (substring.equalsIgnoreCase("music")) {
+							minecraft.sndManager.addMusic(substring2, file2);
+						}
+					}
+					catch (Exception ex) {
+						ex.printStackTrace();
+					}
+				}
+				if (this.closing) {
+					return;
+				}
+			}
+		}
+		catch (IOException ex2) {
+			ex2.printStackTrace();
+		}
 	}
 
 	private void downloadResource(URL var1, File var2) throws IOException {
diff -r -U 3 minecraft/net/minecraft/client/render/EntityRenderer.java minecraft_patched/net/minecraft/client/render/EntityRenderer.java
--- minecraft/net/minecraft/client/render/EntityRenderer.java	2021-11-26 21:41:39.308581900 +0200
+++ minecraft_patched/net/minecraft/client/render/EntityRenderer.java	2021-11-26 21:36:48.086925000 +0200
@@ -359,10 +359,10 @@
 				double var31 = (double)(MathHelper.cos(var54.rotationYaw / 180.0F * 3.1415927F) * MathHelper.cos(var54.rotationPitch / 180.0F * 3.1415927F)) * 4.0D;
 				var33 = (double)(-MathHelper.sin(var54.rotationPitch / 180.0F * 3.1415927F)) * 4.0D;
 
-				for(var45 = (float)false; var45 < 8; ++var45) {
-					float var48 = (float)(((var45 & 1) << 1) - 1);
-					float var42 = (float)(((var45 >> 1 & 1) << 1) - 1);
-					var15 = (float)(((var45 >> 2 & 1) << 1) - 1);
+				for(int var43 = 0; var43 < 8; ++var43) {
+					float var48 = (float)(((var43 & 1) << 1) - 1);
+					float var42 = (float)(((var43 >> 1 & 1) << 1) - 1);
+					var15 = (float)(((var43 >> 2 & 1) << 1) - 1);
 					var48 *= 0.1F;
 					var42 *= 0.1F;
 					var15 *= 0.1F;
diff -r -U 3 minecraft/net/minecraft/client/UnexpectedThrowable.java minecraft_patched/net/minecraft/client/UnexpectedThrowable.java
--- minecraft/net/minecraft/client/UnexpectedThrowable.java	2021-11-26 21:41:39.072568400 +0200
+++ minecraft_patched/net/minecraft/client/UnexpectedThrowable.java	2021-11-26 21:34:40.982655000 +0200
@@ -20,9 +20,6 @@
 		this.exception = var2;
 	}
 
-	public UnexpectedThrowable() {
-	}
-
 	public static NBTTagCompound readCompressed(InputStream var0) throws IOException {
 		DataInputStream var4 = new DataInputStream(new GZIPInputStream(var0));
 
diff -r -U 3 minecraft/net/minecraft/game/item/Item.java minecraft_patched/net/minecraft/game/item/Item.java
--- minecraft/net/minecraft/game/item/Item.java	2021-11-26 21:41:39.191575200 +0200
+++ minecraft_patched/net/minecraft/game/item/Item.java	2021-11-26 21:37:58.761967400 +0200
@@ -142,367 +142,367 @@
 		ItemSpade var10000 = new ItemSpade(0, 2);
 		byte var1 = 82;
 		ItemSpade var0 = var10000;
-		var10000.iconIndex = var1;
+		var10000.setIconIndex(var1);
 		shovel = var0;
 		ItemPickaxe var19 = new ItemPickaxe(1, 2);
 		var1 = 98;
 		ItemPickaxe var2 = var19;
-		var19.iconIndex = var1;
+		var19.setIconIndex(var1);
 		pickaxeSteel = var2;
 		ItemAxe var20 = new ItemAxe(2, 2);
 		var1 = 114;
 		ItemAxe var3 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		axeSteel = var3;
 		ItemFlintAndSteel var21 = new ItemFlintAndSteel(3);
 		var1 = 5;
 		ItemFlintAndSteel var4 = var21;
-		var21.iconIndex = var1;
+		var21.setIconIndex(var1);
 		striker = var4;
 		ItemFood var22 = new ItemFood(4, 4);
 		var1 = 10;
 		ItemFood var5 = var22;
-		var22.iconIndex = var1;
+		var22.setIconIndex(var1);
 		appleRed = var5;
 		ItemBow var23 = new ItemBow(5);
 		var1 = 21;
 		ItemBow var6 = var23;
-		var23.iconIndex = var1;
+		var23.setIconIndex(var1);
 		bow = var6;
 		Item var24 = new Item(6);
 		var1 = 37;
 		Item var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		arrow = var7;
 		var24 = new Item(7);
 		var1 = 7;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		coal = var7;
 		var24 = new Item(8);
 		var1 = 55;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		diamond = var7;
 		var24 = new Item(9);
 		var1 = 23;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		ingotIron = var7;
 		var24 = new Item(10);
 		var1 = 39;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		ingotGold = var7;
 		ItemSword var25 = new ItemSword(11, 2);
 		var1 = 66;
 		ItemSword var8 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		swordSteel = var8;
 		var25 = new ItemSword(12, 0);
 		var1 = 64;
 		var8 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		swordWood = var8;
 		var10000 = new ItemSpade(13, 0);
 		var1 = 80;
 		var0 = var10000;
-		var10000.iconIndex = var1;
+		var10000.setIconIndex(var1);
 		shovelWood = var0;
 		var19 = new ItemPickaxe(14, 0);
 		var1 = 96;
 		var2 = var19;
-		var19.iconIndex = var1;
+		var19.setIconIndex(var1);
 		pickaxeWood = var2;
 		var20 = new ItemAxe(15, 0);
 		var1 = 112;
 		var3 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		axeWood = var3;
 		var25 = new ItemSword(16, 1);
 		var1 = 65;
 		var8 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		swordStone = var8;
 		var10000 = new ItemSpade(17, 1);
 		var1 = 81;
 		var0 = var10000;
-		var10000.iconIndex = var1;
+		var10000.setIconIndex(var1);
 		shovelStone = var0;
 		var19 = new ItemPickaxe(18, 1);
 		var1 = 97;
 		var2 = var19;
-		var19.iconIndex = var1;
+		var19.setIconIndex(var1);
 		pickaxeStone = var2;
 		var20 = new ItemAxe(19, 1);
 		var1 = 113;
 		var3 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		axeStone = var3;
 		var25 = new ItemSword(20, 3);
 		var1 = 67;
 		var8 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		swordDiamond = var8;
 		var10000 = new ItemSpade(21, 3);
 		var1 = 83;
 		var0 = var10000;
-		var10000.iconIndex = var1;
+		var10000.setIconIndex(var1);
 		shovelDiamond = var0;
 		var19 = new ItemPickaxe(22, 3);
 		var1 = 99;
 		var2 = var19;
-		var19.iconIndex = var1;
+		var19.setIconIndex(var1);
 		pickaxeDiamond = var2;
 		var20 = new ItemAxe(23, 3);
 		var1 = 115;
 		var3 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		axeDiamond = var3;
 		var24 = new Item(24);
 		var1 = 53;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		stick = var7;
 		var24 = new Item(25);
 		var1 = 71;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		bowlEmpty = var7;
 		ItemSoup var26 = new ItemSoup(26, 10);
 		var1 = 72;
 		ItemSoup var9 = var26;
-		var26.iconIndex = var1;
+		var26.setIconIndex(var1);
 		bowlSoup = var9;
 		var25 = new ItemSword(27, 0);
 		var1 = 68;
 		var8 = var25;
-		var25.iconIndex = var1;
+		var25.setIconIndex(var1);
 		swordGold = var8;
 		var10000 = new ItemSpade(28, 0);
 		var1 = 84;
 		var0 = var10000;
-		var10000.iconIndex = var1;
+		var10000.setIconIndex(var1);
 		shovelGold = var0;
 		var19 = new ItemPickaxe(29, 0);
 		var1 = 100;
 		var2 = var19;
-		var19.iconIndex = var1;
+		var19.setIconIndex(var1);
 		pickaxeGold = var2;
 		var20 = new ItemAxe(30, 0);
 		var1 = 116;
 		var3 = var20;
-		var20.iconIndex = var1;
+		var20.setIconIndex(var1);
 		axeGold = var3;
 		var24 = new Item(31);
 		var1 = 8;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		silk = var7;
 		var24 = new Item(32);
 		var1 = 24;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		feather = var7;
 		var24 = new Item(33);
 		var1 = 40;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		gunpowder = var7;
 		ItemHoe var27 = new ItemHoe(34, 0);
 		short var11 = 128;
 		ItemHoe var10 = var27;
-		var27.iconIndex = var11;
+		var27.setIconIndex(var11);
 		hoeWood = var10;
 		var27 = new ItemHoe(35, 1);
 		var11 = 129;
 		var10 = var27;
-		var27.iconIndex = var11;
+		var27.setIconIndex(var11);
 		hoeStone = var10;
 		var27 = new ItemHoe(36, 2);
 		var11 = 130;
 		var10 = var27;
-		var27.iconIndex = var11;
+		var27.setIconIndex(var11);
 		hoeSteel = var10;
 		var27 = new ItemHoe(37, 3);
 		var11 = 131;
 		var10 = var27;
-		var27.iconIndex = var11;
+		var27.setIconIndex(var11);
 		hoeDiamond = var10;
 		var27 = new ItemHoe(38, 4);
 		var11 = 132;
 		var10 = var27;
-		var27.iconIndex = var11;
+		var27.setIconIndex(var11);
 		hoeGold = var10;
 		ItemSeeds var28 = new ItemSeeds(39, Block.crops.blockID);
 		var1 = 9;
 		ItemSeeds var12 = var28;
-		var28.iconIndex = var1;
+		var28.setIconIndex(var1);
 		seeds = var12;
 		var24 = new Item(40);
 		var1 = 25;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		wheat = var7;
 		var22 = new ItemFood(41, 5);
 		var1 = 41;
 		var5 = var22;
-		var22.iconIndex = var1;
+		var22.setIconIndex(var1);
 		bread = var5;
 		ItemArmor var29 = new ItemArmor(42, 0, 0, 0);
 		var1 = 0;
 		ItemArmor var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		helmetLeather = var13;
 		var29 = new ItemArmor(43, 0, 0, 1);
 		var1 = 16;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		plateLeather = var13;
 		var29 = new ItemArmor(44, 0, 0, 2);
 		var1 = 32;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		legsLeather = var13;
 		var29 = new ItemArmor(45, 0, 0, 3);
 		var1 = 48;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		bootsLeather = var13;
 		var29 = new ItemArmor(46, 1, 1, 0);
 		var1 = 1;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		helmetChain = var13;
 		var29 = new ItemArmor(47, 1, 1, 1);
 		var1 = 17;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		plateChain = var13;
 		var29 = new ItemArmor(48, 1, 1, 2);
 		var1 = 33;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		legsChain = var13;
 		var29 = new ItemArmor(49, 1, 1, 3);
 		var1 = 49;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		bootsChain = var13;
 		var29 = new ItemArmor(50, 2, 2, 0);
 		var1 = 2;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		helmetSteel = var13;
 		var29 = new ItemArmor(51, 2, 2, 1);
 		var1 = 18;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		plateSteel = var13;
 		var29 = new ItemArmor(52, 2, 2, 2);
 		var1 = 34;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		legsSteel = var13;
 		var29 = new ItemArmor(53, 2, 2, 3);
 		var1 = 50;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		bootsSteel = var13;
 		var29 = new ItemArmor(54, 3, 3, 0);
 		var1 = 3;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		helmetDiamond = var13;
 		var29 = new ItemArmor(55, 3, 3, 1);
 		var1 = 19;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		plateDiamond = var13;
 		var29 = new ItemArmor(56, 3, 3, 2);
 		var1 = 35;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		legsDiamond = var13;
 		var29 = new ItemArmor(57, 3, 3, 3);
 		var1 = 51;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		bootsDiamond = var13;
 		var29 = new ItemArmor(58, 1, 4, 0);
 		var1 = 4;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		helmetGold = var13;
 		var29 = new ItemArmor(59, 1, 4, 1);
 		var1 = 20;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		plateGold = var13;
 		var29 = new ItemArmor(60, 1, 4, 2);
 		var1 = 36;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		legsGold = var13;
 		var29 = new ItemArmor(61, 1, 4, 3);
 		var1 = 52;
 		var13 = var29;
-		var29.iconIndex = var1;
+		var29.setIconIndex(var1);
 		bootsGold = var13;
 		var24 = new Item(62);
 		var1 = 6;
 		var7 = var24;
-		var24.iconIndex = var1;
+		var24.setIconIndex(var1);
 		flint = var7;
 		var22 = new ItemFood(63, 3);
 		var1 = 87;
 		var5 = var22;
-		var22.iconIndex = var1;
+		var22.setIconIndex(var1);
 		porkRaw = var5;
 		var22 = new ItemFood(64, 8);
 		var1 = 88;
 		var5 = var22;
-		var22.iconIndex = var1;
+		var22.setIconIndex(var1);
 		porkCooked = var5;
 		ItemPainting var30 = new ItemPainting(65);
 		var1 = 26;
 		ItemPainting var14 = var30;
-		var30.iconIndex = var1;
+		var30.setIconIndex(var1);
 		painting = var14;
 		var22 = new ItemFood(66, 42);
 		var1 = 11;
 		var5 = var22;
-		var22.iconIndex = var1;
+		var22.setIconIndex(var1);
 		appleGold = var5;
 		ItemSign var31 = new ItemSign(67);
 		var1 = 42;
 		ItemSign var15 = var31;
-		var31.iconIndex = var1;
+		var31.setIconIndex(var1);
 		sign = var15;
 		ItemDoor var32 = new ItemDoor(68);
 		var1 = 43;
 		ItemDoor var16 = var32;
-		var32.iconIndex = var1;
+		var32.setIconIndex(var1);
 		doorWood = var16;
 		ItemBucket var33 = new ItemBucket(69, 0);
 		var1 = 74;
 		ItemBucket var17 = var33;
-		var33.iconIndex = var1;
+		var33.setIconIndex(var1);
 		bucketEmpty = var17;
 		var33 = new ItemBucket(70, Block.waterMoving.blockID);
 		var1 = 75;
 		var17 = var33;
-		var33.iconIndex = var1;
+		var33.setIconIndex(var1);
 		bucketWater = var17;
 		var33 = new ItemBucket(71, Block.lavaMoving.blockID);
 		var1 = 76;
 		var17 = var33;
-		var33.iconIndex = var1;
+		var33.setIconIndex(var1);
 		bucketLava = var17;
 		ItemMinecart var34 = new ItemMinecart(72);
 		var11 = 135;
 		ItemMinecart var18 = var34;
-		var34.iconIndex = var11;
+		var34.setIconIndex(var11);
 		minecartEmpty = var18;
 	}
 }
diff -r -U 3 minecraft/net/minecraft/game/world/path/Pathfinder.java minecraft_patched/net/minecraft/game/world/path/Pathfinder.java
--- minecraft/net/minecraft/game/world/path/Pathfinder.java	2021-11-26 21:41:39.251578600 +0200
+++ minecraft_patched/net/minecraft/game/world/path/Pathfinder.java	2021-11-26 21:40:25.407355000 +0200
@@ -28,7 +28,7 @@
 	private PathEntity createEntityPathTo(Entity var1, double var2, double var4, double var6, float var8) {
 		this.path.clearPath();
 		this.pointMap.clear();
-		Object var9 = this.openPoint(MathHelper.floor_double(var1.boundingBox.minX), MathHelper.floor_double(var1.boundingBox.minY), MathHelper.floor_double(var1.boundingBox.minZ));
+		PathPoint var9 = this.openPoint(MathHelper.floor_double(var1.boundingBox.minX), MathHelper.floor_double(var1.boundingBox.minY), MathHelper.floor_double(var1.boundingBox.minZ));
 		PathPoint var21 = this.openPoint(MathHelper.floor_double(var2 - (double)(var1.width / 2.0F)), MathHelper.floor_double(var4), MathHelper.floor_double(var6 - (double)(var1.width / 2.0F)));
 		PathPoint var3 = new PathPoint(MathHelper.floor_float(var1.width + 1.0F), MathHelper.floor_float(var1.height + 1.0F), MathHelper.floor_float(var1.width + 1.0F));
 		float var24 = var8;
@@ -61,7 +61,7 @@
 			}
 
 			var25.isFirst = true;
-			Object var15 = 0;
+			int var15 = 0;
 			byte var16 = 0;
 			if(var20.findPathOptions(var25.xCoord, var25.yCoord + 1, var25.zCoord, var5) > 0) {
 				var16 = 1;
@@ -88,9 +88,7 @@
 				var20.pathOptions[var15++] = var10;
 			}
 
-			var9 = var15;
-
-			for(int var26 = 0; var26 < var9; ++var26) {
+			for(int var26 = 0; var26 < var15; ++var26) {
 				PathPoint var11 = var20.pathOptions[var26];
 				float var12 = var25.totalPathDistance + var25.distanceTo(var11);
 				if(!var11.isAssigned() || var12 < var11.totalPathDistance) {
diff -r -U 3 minecraft/net/minecraft/game/world/terrain/ChunkProviderGenerate.java minecraft_patched/net/minecraft/game/world/terrain/ChunkProviderGenerate.java
--- minecraft/net/minecraft/game/world/terrain/ChunkProviderGenerate.java	2021-11-26 21:41:39.252578700 +0200
+++ minecraft_patched/net/minecraft/game/world/terrain/ChunkProviderGenerate.java	2021-11-26 21:40:38.628111200 +0200
@@ -371,7 +371,6 @@
 					var3 = var3;
 					var2 = var2;
 					var1 = var1;
-					this = this;
 					continue label204;
 				}
 
diff -r -U 3 minecraft/net/minecraft/game/world/World.java minecraft_patched/net/minecraft/game/world/World.java
--- minecraft/net/minecraft/game/world/World.java	2021-11-26 21:41:39.214576500 +0200
+++ minecraft_patched/net/minecraft/game/world/World.java	2021-11-26 21:40:33.997846300 +0200
@@ -961,7 +961,6 @@
 			}
 
 			var1 = var1.riddenByEntity;
-			this = this;
 		}
 	}
 
